api/views.py

```diff
@@ -40,31 +40,30 @@ def post(self, request):
         Return a query set according to the post message status.
         """
         try:
-            request_type = request.data['type']
-            size = request.data['limit']
+            request_type = request.data.get('type')
+            size = request.data.get('limit', None)
 
             if request_type == "brownbag":
-                # Get all users IDs elligible for brownbag
-                users_queryset = User.objects.filter(
-                    profile__brownbag="not_done").values_list('id', flat=True)
-                users = list(users_queryset)
-                rand = Randomizer(users)
-                next_brownbag_user = rand.get_random()
-                today = datetime.date.today()
-                weekday = today.weekday()
-                next_friday = today + datetime.timedelta((4 - weekday) % 7)
-                data = {
-                    "date": str(next_friday),
-                    "status": "next_in_line",
-                    "user": next_brownbag_user
-                }
-                serializer = BrownbagSerializer(data=data)
-                if serializer.is_valid():
-                    serializer.save()
+                try:
```

Comment:
Would it be possible to refactor this to get rid of the nested `try...except` blocks?
Maybe something similar to
```python
try:
    # import pdb; pdb.set_trace()
    brownbag_data = create_brownbag(datetime.date.today())
except IntegrityError:
    # There exists a brownbag this Friday, create for next one
    latest_brownbag = Brownbag.objects.latest('date')
    # Use latest friday as seed date to create next brownbag
    new_brownbag = create_brownbag(latest_brownbag.date)
except Exception as e:
    return Response(
        {'error': e}, status=status.HTTP_400_BAD_REQUEST)
else:
    return Response(
        brownbag_data, status=status.HTTP_201_CREATED)
```

**Note:** I haven't tried that yet, it might fail catastrophically and implode the universe. 😄